scoring_board.cpp

```diff
@@ -0,0 +1,93 @@
+#include "scoring_board.h"
+#include <cassert>
+#include <cmath>
+#include <fstream>
+#include <string>
+
+scoring_board::scoring_board(unsigned int score_player1,
+                             unsigned int score_player2 ,
+                             unsigned int score_player3,
+                             unsigned int timer,
+                             short winner)
+    : m_score_player1{score_player1},
+      m_score_player2{score_player2},
+      m_score_player3{score_player3},
+      m_timer{timer},
+      m_winner{winner}
+{
+
+}
+
+void scoring_board::set_winner(short winner)
+{
+  if ((winner < 0) | (winner > 2))
+    {
+      throw std::logic_error("winner should only be 0, 1 or 2");
+    } else {
+      m_winner = winner;
+    }
+}
+
+void test_scoring_board()
+{
+#ifndef NDEBUG // no tests in release
+    #define FIX_ISSUE_542
+    #ifdef FIX_ISSUE_542
+    // Scoring board has a correct initial state
+    {
+        scoring_board sb;
+        assert(sb.get_score_player1() == 0);
+        assert(sb.get_score_player2() == 0);
+        assert(sb.get_score_player3() == 0);
+        assert(sb.get_timer() == 0);
+        assert(sb.get_winner() == -1);
```

Comment:
What we could do is make the winner an enum class that can take the value 0, 1, 2 or none.

Reply:
#590